File: src/main/java/com/flexpoker/table/query/handlers/PotAmountIncreasedEventHandler.java
package com.flexpoker.table.query.handlers;

import java.util.Set;
import java.util.stream.Collectors;

import javax.inject.Inject;

import org.springframework.stereotype.Component;

import com.flexpoker.framework.event.EventHandler;
import com.flexpoker.framework.pushnotifier.PushNotification;
import com.flexpoker.framework.pushnotifier.PushNotificationPublisher;
import com.flexpoker.pushnotifications.TableUpdatedPushNotification;
import com.flexpoker.table.command.events.PotAmountIncreasedEvent;
import com.flexpoker.table.query.repository.TableRepository;
import com.flexpoker.web.dto.outgoing.PotDTO;
import com.flexpoker.web.dto.outgoing.TableDTO;

@Component
public class PotAmountIncreasedEventHandler implements
        EventHandler<PotAmountIncreasedEvent> {

    private final TableRepository tableRepository;

    private final PushNotificationPublisher pushNotificationPublisher;

    @Inject
    public PotAmountIncreasedEventHandler(TableRepository tableRepository,
            PushNotificationPublisher pushNotificationPublisher) {
        this.tableRepository = tableRepository;
        this.pushNotificationPublisher = pushNotificationPublisher;
    }

    @Override
    public void handle(PotAmountIncreasedEvent event) {
        handleUpdatingTable(event);
        handlePushNotifications(event);
    }

    private void handleUpdatingTable(PotAmountIncreasedEvent event) {
        TableDTO currentTable = tableRepository.fetchById(event.getAggregateId());

        System.out.println("increase: " + event.getAmountIncreased());

        Set<PotDTO> pots = currentTable.getPots().stream()
                .map(x -> x.isOpen()
                        ? new PotDTO(x.getSeats(), x.getAmount() + event.getAmountIncreased(), x.isOpen(), x.getWinners())
                        : new PotDTO(x.getSeats(), x.getAmount(), x.isOpen(), x.getWinners()))
                .collect(Collectors.toSet());

        TableDTO updatedTable = new TableDTO(currentTable.getId(),
                event.getVersion(), currentTable.getSeats(),
                currentTable.getTotalPot(), pots,
                currentTable.getVisibleCommonCards(),
                currentTable.getCurrentHandMinRaiseToAmount());
        tableRepository.save(updatedTable);
    }

    private void handlePushNotifications(PotAmountIncreasedEvent event) {
        PushNotification pushNotification = new TableUpdatedPushNotification(
                event.getGameId(), event.getAggregateId());
        pushNotificationPublisher.publish(pushNotification);
    }

}

File: src/main/kotlin/com/flexpoker/table/query/handlers/PotAmountIncreasedEventHandler.kt
package com.flexpoker.table.query.handlers

import com.flexpoker.framework.event.EventHandler
import com.flexpoker.framework.pushnotifier.PushNotificationPublisher
import com.flexpoker.pushnotifications.TableUpdatedPushNotification
import com.flexpoker.table.command.events.PotAmountIncreasedEvent
import com.flexpoker.table.query.repository.TableRepository
import org.springframework.stereotype.Component
import javax.inject.Inject

@Component
class PotAmountIncreasedEventHandler @Inject constructor(
    private val tableRepository: TableRepository,
    private val pushNotificationPublisher: PushNotificationPublisher
) : EventHandler<PotAmountIncreasedEvent> {

    override fun handle(event: PotAmountIncreasedEvent) {
        handleUpdatingTable(event)
        handlePushNotifications(event)
    }

    private fun handleUpdatingTable(event: PotAmountIncreasedEvent) {
        val tableDTO = tableRepository.fetchById(event.aggregateId)
        val updatedPots = tableDTO.pots!!
            .map {
                if (it.isOpen) {
                    it.copy(amount = it.amount + event.amountIncreased)
                } else {
                    it
                }
            }.toSet()
        val updatedTable = tableDTO.copy(version = event.version, pots = updatedPots)
        tableRepository.save(updatedTable)
    }

    private fun handlePushNotifications(event: PotAmountIncreasedEvent) {
        val pushNotification = TableUpdatedPushNotification(event.gameId, event.aggregateId)
        pushNotificationPublisher.publish(pushNotification)
    }

}